// lucii/lib/util/config.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _fs = require('fs');

var _path = require('path');

var _stripJsonComments = require('strip-json-comments');

var _stripJsonComments2 = _interopRequireDefault(_stripJsonComments);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
    var config = JSON.parse((0, _stripJsonComments2.default)((0, _fs.readFileSync)('.lucii', 'utf-8')));
    var html = config.html;

    return (0, _extends3.default)({
        publicPath: '/',
        outputPath: './dist',
        host: 'localhost',
        port: 8352
    }, config, {
        html: {
            title: html.title,
            favicon: html.favicon,
            template: html.template,
            filename: html.filename || (0, _path.parse)(html.template).base
        }
    });
};

module.exports = exports['default'];
