// lucii/demo/vue/proxy.js
/**
 * @file config
 */
const mockDomain = 'localhost',
    mockPort = 3001,
    serverConfig = {
        JingYong: '10.99.23.19:8982'
    };
let mockServer = `http://${mockDomain}:${mockPort}`;
module.exports = {
    '/post2': {
        target: mockServer,
    },
    // 联调地址 Cookie 需要实时更新（痛点）
    '/api/framework': {
        target: mockServer,
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },
    '/api/upload': {
        target: mockServer
    },
    '/api/humgroup': {
        target: mockServer
    },
    '/api/experiment/exp/fuzzyquery': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080',
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },
    '/api/experiment/expplan/planedit': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080',
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },
    // 人群包模糊查询
    '/api/crowd/expcrowd/search': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 风控策略模糊查询
    '/api/strategy/riskstra/query': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 保存实验
    '/api/experiment/expplan/planstracreate': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },

    // 实验导出
    '/api/experiment/export/*': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },

    // 实验管理列表
    '/api/experiment/exp/*': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },

    // 实验计划搜索
    '/api/experiment/expplan/plansearch': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },
    // 入白页上一页
    '/api/experiment/exp/prepage': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true
    },

    // 触达
    '/api/bosstouch/touch/begin': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },

    // 保存实验2
    '/api/experiment/expplan/planstrasave': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 生成实验
    '/api/experiment/exp/expcreate': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    '/api/experiment/exp/samplesave': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 开始抽样
    '/api/experiment/exp/samplestart': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 推风控及入白
    '/api/experiment/exp/whiteliststart': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 获取产品列表
    '/api/crowd/*': {
        target: `http://${serverConfig.JingYong}`,
        secure: false,
        changeOrigin: true,
        hostRewrite: serverConfig.JingYong,
        headers: {
            Host: serverConfig.JingYong,
            /* eslint-disable */
            Cookie: 'BAIDUID=B61FE0F0BCDFDC6B888CE3A6D464B498:FG=1; BIDUPSID=B61FE0F0BCDFDC6B888CE3A6' +
                    'D464B498; PSTM=1490200068; locale=zh; BDUSS=JNcXY1SH42MVE4eS1WZ0J-eEhoRFZIQUNoa2' +
                    'R4Y2UtcVhhNjR3a1ZGR29qQTFaSUFBQUFBJCQAAAAAAAAAAAEAAADFiQeVYm9zdGVzdDAxAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKj~5Vio~-VYeF; pg' +
                    'v_pvi=4325972992; pgv_si=s9151893504; PSINO=5; H_PS_PSSID=1453_13551_21102_17001' +
                    '_20719; BDORZ=B490B5EBF6F3CD402E515D22BCDA1598; PHPSESSID=ST-76170-iiliZSheUU0cS' +
                    'drbAcHr-uuap'
            /* eslint-enable */
        }
    },

    '/api/strategy/customergathering': {
        hostRewrite: '10.152.77.42:8080',

        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        headers: {
            Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
            /* eslint-disable */
            Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                    'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                    'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                    'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                    '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                    '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                    '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                    'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                    '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                    'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                    'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                    'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                    '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                    'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                    'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                    'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                    'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                    'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                    'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                    'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                    'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                    'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                    '2; H_PS_PSSID=1439_21091_17001_22158'
            /* eslint-enable */
        }
    },
    '/api/strategy/product': {
        target: 'http://10.152.77.42:8080',
        secure: false,
        changeOrigin: true,
        hostRewrite: '10.152.77.42:8080'
    },
    // 效果数据和权限管理模块proxy转发
    '/api/statistics': {
        target: 'http://cq01-rdqa-dev059.cq01.baidu.com:8810/',
        secure: false,
        changeOrigin: true,
        headers: {
            target: 'http://10.152.77.42:8080',
            secure: false,
            changeOrigin: true,
            hostRewrite: '10.152.77.42:8080',
            headers: {
                Host: 'cq01-rdqa-dev059.cq01.baidu.com:8810',
                /* eslint-disable */
                Cookie: 'BAIDUID=94AA6D53EAF771FA63F58BE0F778BBF7:FG=1; BIDUPSID=94AA6D53EAF771FA63F58BE0' +
                        'F778BBF7; PSTM=1481459478; BDUSS=Ixckh1cnk2Q2x6Mmk4WEZ-fnhESldnUWc3eDZ-c2prb2dCd' +
                        'XhHWk9KaGlKS0JZSVFBQUFBJCQAAAAAAAAAAAEAAACyGAoId29sZnFzYwAAAAAAAAAAAAAAAAAAAAAAA' +
                        'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGKXeFhil3hYaH; H_WISE_SIDS=1' +
                        '14454_102572_104494_100042_114516_113890_114177_104381_100099_106435_114000_1121' +
                        '06_107312_112135_114131_114098_114797_114513_114287_114329_114535_114314_114276_' +
                        '114718_114078_114491_114490_110085_100459; MCITY=-334%3A; BDSFRCVID=4lAsJeC62R7a' +
                        'w13iIkGjMpkflMvfxA7TH6aoa9qZOZzlcP5nuE3OEG0PfM8g0KubVI2-ogKKyeOTHubP; H_BDCLCKID' +
                        '_SF=fnID_K-Kf-3bfTrg-tcHMDCShGRaQfJeWDTm_DoEW-5zHR7Y-T30MRQbyPniKKrQKmvq-pPKKR78' +
                        'oKOHj6QlKJQBQbDfJtDf3mkjbPbDfn02OPKzehnNhP4syP4j2xRnWTcabIFXfC-ahC-RenJb5ICbMqb3' +
                        'aRotHD7yWCvSHlv5OR5Jj65nyjF0Qpj055OIJ6IOah7c5h82V-8R3MA--t4d-JbtB5bK56Kj24Db0b3E' +
                        'sq0x0bQYe-bQyp_LKUoDHCOMahkMal7xO-QP05CKDTo0ja0jtTnXa5n-stTObRbHDTrnhPF3Ll8UKP6-' +
                        '35KHMgn4BRbF2ROUfb6GeMRcyP-UDl7Eaq37JD6yQj6o54O48PTq0-OM3xoQQ4oxJpODBRbMopvaKJjs' +
                        'MxcvbURvD-ug3-7q5q8EtR-OVI_MtK-3fP36q4vEbJ8thmT22-usWKnRQhcH0KLKsPoOWfncMt0Tqqjt' +
                        'qUn9WKvi3CJntMb1MRnDb-b_0-A0hJJU0tbDfT7a3l5TtUJa8DnTDMRh-xk8QnQyKMnitIj9-pnKKft0' +
                        'hI8wD6uaj6P3KxQ2a-QHHDjKQJT8-bC_Hn7zeURZebtpbt-qJJjIWaKD-nTw2Undfho5b63JyTLqWGon' +
                        'BT5KaaAf-InntqOnMxQeBp5SD5-kQN3TKJkO5bRiL66hbKoADn3oyT3VXp0njGoTqjKffnFH_C8QKbRE' +
                        'Hnkk-tvE2JLHqxby26nrt5TeaJ5n0-nnhn6jbPTEytPjhq8OQnFHbN8LaJb_3pTIsI8Ry6CajTcWDaDD' +
                        'q6n2aI52B5rqanTBfJuk-PnVeT0jLqbZKxtqtHuf340htbTzjp3FXJ3Ob4AI5xjO36bnWncKWbDX-xcc' +
                        'o4c1-p0hhRLkDPn405OT5eDO0KJcbRo4OD_zhPJvynF8XnO7-4jTfD7H3KCaJDIhbMK; BKMASKSID=d' +
                        'a00b87bde806533c15af0efbbc664f8; PHPSESSID=ST-3510-sZBtg1bxwVKKk9K5tdba-uuap; BD' +
                        'ORZ=B490B5EBF6F3CD402E515D22BCDA1598; BDRCVFR[feWj1Vr5u3D]=I67x6TjHwwYf0; PSINO=' +
                        '2; H_PS_PSSID=1439_21091_17001_22158'
                /* eslint-enable */
            }
        }
    },
    '/api/system': {
        target: mockServer
    }
    // 新建实验计划
};
