// lucii/lib/config/webpack.config.dll.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

exports.default = function (config, pkg) {
  var dependencies = (0, _keys2.default)(pkg.dependencies);
  var vendors = config.commons || dependencies;
  console.log(vendors);
  return {
    entry: {
      vendor: vendors
    },
    output: {
      path: _path2.default.resolve(config.outputPath),
      filename: '[name].js',
      library: '[name]'
    },
    plugins: [new _webpack2.default.DllPlugin({ path: 'manifest.json', name: '[name]', context: config.root })]
  };
};

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

var _webpack = require('webpack');

var _webpack2 = _interopRequireDefault(_webpack);

var _htmlWebpackPlugin = require('html-webpack-plugin');

var _htmlWebpackPlugin2 = _interopRequireDefault(_htmlWebpackPlugin);

var _path3 = require('../util/path');

var _path4 = _interopRequireDefault(_path3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

;
module.exports = exports['default'];
