// lucii/src/config/webpack.config.prop.js
import {resolve} from 'path';
import webpack from 'webpack';
import autoprefixer from 'autoprefixer';
import HtmlWebpackPlugin from 'html-webpack-plugin';
import ExtractTextPlugin from 'extract-text-webpack-plugin';
import getPath from '../util/path';

export default function (config, pkg) {
    const NODE_ENV = process.env.NODE_ENV;
    const dependencies = Object.keys(pkg.dependencies);
    const vendors = config.commons || dependencies;
    const getThemeConfig = require(getPath(process.cwd())(config.theme));
    const theme = getThemeConfig(config.publicPath);
    const {title, favicon, template, filename} = config.html;
    const plugins = [
        new webpack.optimize.UglifyJsPlugin()
    ];
    return {
        entry: {...config.entry,
            vendor: vendors
        },
        output: {
            filename: '[name].[hash:5].js',
            publicPath: config.publicPath,
            path: resolve(config.outputPath),
            chunkFilename: 'js/[name].[chunkhash:5].js'
        },
        resolveLoader: {
            moduleExtensions: ['-loader'],
        },
        resolve: {
            extensions: ['.js', '.vue', '.json'],
            alias: {
                'react/lib/ReactMount': 'react-dom/lib/ReactMount',
                'vue$': 'vue/dist/vue.esm.js',
                '@': resolve('src')
            }
        },
        module: {
            rules: [{
                    test: /\.vue$/,
                    loader: ['vue-loader']
                }, {
                    test: /\.jsx$/,
                    loaders: ['babel']
                }, {
                    test: /\.js?$/,
                    exclude: [/node_modules/],
                    loaders: ['babel']
                    // loaders: ['react-hot', 'babel']
                }, {
                    test: /\.json$/,
                    loader: 'json'
                }, {
                    test: /\.css$/,
                    loader: ExtractTextPlugin.extract({
                        fallback: 'style',
                        use: `css?{"modifyVars":${JSON.stringify(theme)}}`
                    })
                }, {
                    test: /\.less$/,
                    include: [
                        /node_modules/, /index\.less$/
                    ],
                    loader: ExtractTextPlugin.extract({
                        fallback: 'style',
                        use: `css?importLoaders=1!postcss!less?{"modifyVars":${JSON.stringify(theme)}}`
                    })
                }, {
                    test: /\.less$/,
                    exclude: [
                        /node_modules/, /index\.less$/
                    ],
                    loader: ExtractTextPlugin.extract({
                        fallback: 'style',
                        use: 'css?importLoaders=1&modules&localIdentName=[name]__[local]!postcss!less'
                    })
                }, {
                    test: /\.(png|gif|jpg|svg|ico)$/,
                    loader: 'url',
                    options: {
                        limit: 10000,
                        name: 'static/[name].[hash:5].[ext]'
                    }
                }, {
                    test: /\.(woff|eot|ttf)\??.*$/,
                    loader: 'url',
                    options: {
                        limit: 10000,
                        name: 'static/[name].[ext]'
                    }
                }
            ]
        },
        plugins: [
            new webpack.LoaderOptionsPlugin({
                options: {
                    babel: {
                        babelrc: false,
                        presets: [
                            require.resolve('babel-preset-es2015'),
                            require.resolve('babel-preset-react'),
                            require.resolve('babel-preset-stage-0'),
                        ],
                        plugins: [
                            // require.resolve('babel-plugin-add-module-exports'),
                            // require.resolve('babel-plugin-react-require'),
                        ].concat(config.production.extraBabelPlugins || []),
                        cacheDirectory: true
                    },
                    postcss() {
                        return {
                            defaults: [autoprefixer],
                            cleaner: [autoprefixer({browsers: ['last 4 versions']})]
                        };
                    }
                }
            }),
            new webpack.DefinePlugin({
                'process.env': {
                    NODE_ENV: JSON.stringify(NODE_ENV),
                },
            }),
            new HtmlWebpackPlugin({
                title: title,
                favicon: favicon,
                filename: filename,
                template: template,
                inject: true
            }),
            new webpack.optimize.CommonsChunkPlugin({
                name: "vendor",
                filename: "[name].[hash:5].js"
            }),
            new ExtractTextPlugin('[name].[chunkhash:5].css')
        ].concat(plugins)
    }
};
